symengine/polynomial.cpp

```diff
@@ -314,70 +316,48 @@ unsigned int bit_length(T t)
     return count;
 }
 
-RCP<const UnivariateIntPolynomial>
-mul_poly(RCP<const UnivariateIntPolynomial> a,
-         RCP<const UnivariateIntPolynomial> b)
+RCP<const UnivariateIntPolynomial> mul_poly(const UnivariateIntPolynomial &a,
+                                            const UnivariateIntPolynomial &b)
 {
-    // TODO: Use `const RCP<const UnivariateIntPolynomial> &a` for input
-    // arguments,
-    //      even better is use `const UnivariateIntPolynomial &a`
-    RCP<const Symbol> var = symbol("");
-    if (a->get_var()->get_name() == "") {
-        var = b->get_var();
-    } else if (b->get_var()->get_name() == "") {
-        var = a->get_var();
-    } else if (!(a->get_var()->__eq__(*b->get_var()))) {
-        throw std::runtime_error("Error: variables must agree.");
-    } else {
-        var = a->get_var();
-    }
```

Comment:
It is needed. There can be polynomials with no var, just a coefficient which can be used later.
I think we need to rethink the design, but that's not related to this PR.